lsp-serenata.el

```diff
@@ -39,21 +40,58 @@ It can be downloaded from https://gitlab.com/Serenata/Serenata/-/releases."
   :group 'lsp-serenata
   :type 'file)
 
+(defcustom lsp-serenata-php-version
+  7.3
+  "Allows you to specify the PHP version your project is written in.
+At the moment this directive is still ignored, but it will
+influence functionality such as refactoring in the future, where
+older PHP versions may not support scalar type hints, which may
+then be omitted from places such as getters and setters."
+  :group 'lsp-serenata
+  :type 'number)
+
+(defcustom lsp-serenata-file-extensions
+  ["php"]
+  "List of file extensions (without dot) to process.
+Files that do not match this whitelist will be ignored during
+indexing.  Usually you'll want to set this to at least include
+php, as it is the most common PHP extension.  phpt is not
+included by default as it is often used to contain test code that
+is not directly part of the code.  Note that for existing
+projects, removing extensions will not not automatically prune
+files having them from the index if they are already present.
+Adding new ones will cause the files having them to be picked up
+on the next project initialization."
+  :group 'lsp-serenata
+  :type 'lsp-string-array)
+
 (defun lsp-serenata-server-start-fun (port)
   "Define serenata start function, it requires a PORT."
   `(,lsp-serenata-server-path
     "-u" ,(number-to-string port)))
 
+(defun lsp-serenata-init-options ()
+  "Init options for lsp-serenata."
+  `(
+    :phpVersion ,lsp-serenata-php-version
+    :fileExtensions ,lsp-serenata-file-extensions))
+
+(lsp-interface (serenata:didProgressIndexing (:info) nil))
```

Comment:
The type for `serenata/didProgressIndexing` is different from what you put here
https://gitlab.com/Serenata/Serenata/-/blob/master/src/Indexing/DirectoryIndexRequestDemuxer.php

Could you consider to properly add all the type as defined in the source?
It will save other time to modify the interface again

Reply:
Sure, no problem